dds/DCPS/XTypes/TypeObject.h

```diff
@@ -1775,70 +1759,65 @@ namespace XTypes {
   OpenDDS_Dcps_Export
   TypeIdentifier makeTypeIdentifier(const TypeObject& type_object);
 
-  OpenDDS_Dcps_Export
-  void serialize_type_info(const TypeInformation& type_info, DDS::OctetSeq& seq);
-
-  OpenDDS_Dcps_Export
-  void deserialize_type_info(TypeInformation& type_info, const DDS::OctetSeq& seq);
-
-} // namespace XTypes
-
-namespace DCPS {
-
-template<typename T>
-const XTypes::TypeObject& getMinimalTypeObject();
-
-template<typename T>
-XTypes::TypeIdentifier getMinimalTypeIdentifier();
-
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<void>();
-
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::Boolean>();
-
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::Octet>();
-
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::Short>();
+  template <typename T>
+  void serialize_type_info(const TypeInformation& type_info, T& seq)
+  {
+    seq.length(DCPS::serialized_size(XTypes::get_typeobject_encoding(), type_info));
+    DCPS::MessageBlockHelper<T> helper(seq);
+    DCPS::Serializer serializer(helper, XTypes::get_typeobject_encoding());
+    if (!(serializer << type_info)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) serialize_type_info ")
+                 ACE_TEXT("serialization of type information failed.\n")));
+    }
+  }
 
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::Long>();
+  template <typename T>
+  void deserialize_type_info(TypeInformation& type_info, const T& seq)
+  {
+    DCPS::MessageBlockHelper<T> helper(seq);
+    DCPS::Serializer serializer(helper, XTypes::get_typeobject_encoding());
+    if (!(serializer >> type_info)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) deserialize_type_info ")
+                 ACE_TEXT("deserialization of type information failed.\n")));
+    }
+  }
 
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::LongLong>();
+  OpenDDS_Dcps_Export
+  ACE_CDR::ULong hash_member_name_to_id(const std::string& name);
 
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::UShort>();
+  OpenDDS_Dcps_Export
+  void hash_member_name(NameHash& name_hash, const std::string& name);
 
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::ULong>();
+  OpenDDS_Dcps_Export
+  bool is_fully_descriptive(const TypeIdentifier& ti);
 
-template<> OpenDDS_Dcps_Export
-XTypes::TypeIdentifier getMinimalTypeIdentifier<ACE_CDR::ULongLong>();
+  typedef std::map<TypeIdentifier, TypeObject> TypeMap;
```

Comment:
In general I would suggest using the allocator-aware instantiations of the STL containers from the start.  At this point maybe we need to do it as a separate pass.